plugins/woocommerce/includes/admin/helper/class-wc-helper-updater.php

```diff
@@ -157,10 +157,43 @@ public static function transient_update_themes( $transient ) {
 	public static function setup_update_plugins_messages() {
 		foreach ( WC_Helper::get_local_woo_plugins() as $plugin ) {
 			$filename = $plugin['_filename'];
-			add_action( 'in_plugin_update_message-' . $filename, array( __CLASS__, 'add_install_marketplace_plugin_message' ), 10, 2 );
+			if ( WC_Helper::is_site_connected() ) {
+				add_action( 'in_plugin_update_message-' . $filename, array( __CLASS__, 'add_install_marketplace_plugin_message' ), 10, 2 );
+			} else {
+				add_action( 'in_plugin_update_message-' . $filename, array( __CLASS__, 'add_connect_woocom_plugin_message' ) );
+			}
 		}
 	}
 
+	/**
+	 * Runs on in_plugin_update_message-{file-name}, show a message to connect to Woo.com for unconnected stores
+	 *
+	 * @return void.
+	 */
+	public static function add_connect_woocom_plugin_message() {
```

Comment:
on my setup, somehow I couldn't get this callback triggered when visiting `/wp-admin/plugins.php`. It's the same as `add_install_marketplace_plugin_message`. Any ideas?

just a note, i put debug line on the callback and it never reached the line.

Reply:
you may need to install a new different plugin as the update data is cached by the hash value of list of woo plugins